packages/@react-spectrum/s2/src/Picker.tsx

```diff
@@ -394,26 +394,35 @@ export const Picker = /*#__PURE__*/ (forwardRef as forwardRefType)(function Pick
                 estimatedHeadingHeight: 50,
                 padding: 8,
                 loaderHeight: LOADER_ROW_HEIGHTS[size][scale]}}>
-              <PopoverBase
+              <Popover
                 hideArrow
                 offset={menuOffset}
+                crossOffset={isQuiet ? -12 : undefined}
                 placement={`${direction} ${align}` as Placement}
                 shouldFlip={shouldFlip}
                 UNSAFE_style={{
-                  width: menuWidth && !isQuiet ? `${menuWidth}px` : undefined
+                  width: menuWidth && !isQuiet ? `calc(${menuWidth}px - 2 * var(--s2-container-border-width))` : undefined
```

Comment:
can probably use border box here as well

looking ahead, i think that'd fix a lot of these